internal/events/event.go

```diff
@@ -230,21 +205,73 @@ func (e Events) hostEventsQuery(events *[]*common.Event, selectedCategories []st
 		Find(events, "cluster_id = ? AND host_id = ?", clusterID.String(), (*hostID).String())
 }
 
-func (e Events) eventsQuery(events *[]*common.Event, selectedCategories []string, clusterID *strfmt.UUID, hostID *strfmt.UUID, infraEnvID *strfmt.UUID) *gorm.DB {
+func (e Events) eventsQuery(ctx context.Context, events *[]*common.Event, selectedCategories []string, clusterID *strfmt.UUID, hostID *strfmt.UUID, infraEnvID *strfmt.UUID) *gorm.DB {
 	whereCondition := make([]string, 0)
+	user := ocm.UserNameFromContext(ctx)
+
 	if clusterID != nil {
-		whereCondition = append(whereCondition, fmt.Sprintf("cluster_id = '%s'", clusterID.String()))
+		cluster, err := common.GetClusterFromDB(e.db, *clusterID, common.UseEagerLoading)
+		if err != nil {
+			return &gorm.DB{Error: err}
```

Comment:
Please ensure that the same code and message are returned here and in line 224 (where the cluster belongs to another user).
For example, here, you can check if the error is gorm.ErrRecordNotFound, and if so, return the same thing as line 224.

(Same for host and infra-env)

Reply:
Done 👍